apps/site/assets/ts/schedule/components/ScheduleFinder.tsx

```diff
@@ -0,0 +1,204 @@
+    : null;
+
+interface SelectContainerProps {
+  children: ReactNode;
+  error: boolean;
+}
+
+const SelectContainer = ({
+  children,
+  error
+}: SelectContainerProps): ReactElement<HTMLElement> => (
+  <div className={`schedule-finder__select-container ${error && "error"}`}>
+    {children}
+    <div
+      className="schedule-finder__arrow"
+      dangerouslySetInnerHTML={{ __html: "&#xF107;" }}
+    />
+  </div>
+);
+
+interface ErrorMessageProps {
+  directionError: boolean;
+  originError: boolean;
+}
+
+const ErrorMessage = ({
+  directionError,
+  originError
+}: ErrorMessageProps): ReactElement<HTMLElement> | null => {
+  if (!directionError && !originError) {
+    return null;
+  }
+
+  const message =
+    directionError && originError
+      ? "direction and origin"
+      : directionError
+      ? "direction"
+      : "origin";
+
+  return (
+    <div className="error-container">
+      <span role="alert">Choose a value for {message}</span>
+    </div>
+  );
+};
+
+const ScheduleFinder = ({
+  directionDestinations,
+  directionNames,
+  stops
+}: Props): ReactElement<HTMLElement> => {
+  const directionEl = React.useRef<HTMLSelectElement>(null);
+  const stopEl = React.useRef<HTMLSelectElement>(null);
+  const submitButtonEl = React.useRef<HTMLInputElement>(null);
+
+  const [state, setState] = useState<State>({
+    selectedDirection: null,
+    selectedOrigin: null,
+    directionError: false,
+    originError: false
+  });
+
+  const handleClickSubmit = (e: SyntheticEvent): void => {
+    e.preventDefault();
+    const selectedDirection = getSelectedDirection(directionEl);
+    const selectedOrigin = getSelectedOrigin(stopEl);
+
+    if (selectedDirection === null || selectedOrigin === null) {
+      e.stopPropagation(); // prevent modal from opening
+      setState({
+        selectedDirection: null,
+        selectedOrigin: null,
+        directionError: selectedDirection === null,
+        originError: selectedOrigin === null
+      });
+      return;
+    }
+
+    setState({
+      selectedDirection,
+      selectedOrigin,
+      directionError: false,
+      originError: false
+    });
+  };
+
+  return (
+    <div className="schedule-finder">
+      <h2 className="h3 schedule-finder__heading">Schedule Finder</h2>
+      <ErrorMessage
+        directionError={state.directionError}
+        originError={state.originError}
+      />
+      <div>
+        Choose a stop to get schedule information and real-time departure
+        predictions.
+      </div>
+      <SelectContainer error={state.directionError}>
+        <select
+          ref={directionEl}
+          className="schedule-finder__select"
+          onKeyUp={e =>
+            handleReactEnterKeyPress(e, () => {
+              submitButtonEl!.current!.click();
+            })
+          }
+        >
+          <option value="">Choose a direction</option>
+          <option value="0">
+            {directionNames[0]}  {directionDestinations[0]}
```

Comment:
For some reason this causes some weird whitespace characters on SSR. Using a template string prevents it.

```suggestion
            {`${directionNames[0]}  ${directionDestinations[0]}`}
```

Reply:
ahh interesting. For the space between the words I am using a special unicode character to achieve a wider space -- I think that is the cause... but I will try this